### PR TITLE
New git file finder that support symlinks.

### LINKED ISSUE #245: git file finder: follow symlinks
Imagine the following directory structure where setup.py is in a subdirectory in the git repo, and contains a symlink to a parent directory.

```
package/module.py
setup/setup.py
setup/package -> ../package
```

Currently, the git file finder will ignore the symlinked directory as it does a `git ls-files <setup root>`.

Such a directory structure is useful in situations where one does not have freedom to arrange a canonical directory structure in the repo.

Would you review a PR that makes the git file finder follow symlinks (of course restricting links to files/directories that are themselves part of the same git repo).

### PR DESCRIPTION
**Pull request by sbidoul**
Closes #245

### CODE REVIEW

**Comment by RonnyPfannschmidt** [comment]
it seems this breaks the expectations of some tests which is ok,
part of it is pip 10.x breaking the world for bad code (and i had some of that, sorry)

**Comment by sbidoul** [comment]
No need to be sorry about anything. The work behing done here is far from easy, with huge expectations from the world.

It's green now.

File: setuptools_scm/git_file_finder.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Copyright © 2018 ACSONE SA/NV
+# License LGPLv3 (http://www.gnu.org/licenses/lgpl-3.0-standalone.html)
```
Comment (RonnyPfannschmidt) [inline setuptools_scm/git_file_finder.py]: i just noted the licence, is it acceptable for you to go to MIT?

Comment (sbidoul) [reply setuptools_scm/git_file_finder.py]: yep, sorry, copy/paste error, I remove that

**Review by RonnyPfannschmidt — COMMENTED** [review]
generally well done, this is how i like a change :grinning: :+1: 

please take a look at the licensing detail

**Review by RonnyPfannschmidt — APPROVED** [review]

**Comment by RonnyPfannschmidt** [comment]
wow, that's quick ^^

**Comment by sbidoul** [comment]
Thanks.

Is appveyor running tests on windows? I don't find the link.

**Comment by RonnyPfannschmidt** [comment]
i need to check to see why its not there

**Comment by RonnyPfannschmidt** [comment]
it seems a configuration issue is preventing the ci run, it'll take a bit to investigate

**Comment by RonnyPfannschmidt** [comment]
indeed, it got strangely messed up in a token cleanup and appveyor never told details until i recreated the project

**Comment by RonnyPfannschmidt** [comment]
it seems like windows is giving and giving :disappointed:

**Comment by sbidoul** [comment]
@RonnyPfannschmidt a bit of trial and error indeed :)

I think I nailed it now. Basically it boils down to using normcase everywhere when looking up in the git files list. 

I chose to preserve case and separators in the finder output.

The last two failing tests also fail on master.

**Comment by RonnyPfannschmidt** [comment]
thanks for finalizing the details, i'll take a look at the rest

**Comment by RonnyPfannschmidt** [comment]
ok, as far as i understand this is merge-able and the tests that dont work in windows are my fault due to developing while windows ci was accidentally off

please ack then i'll merge

**Comment by sbidoul** [comment]
yes, this PR is ready to merge

**Comment by RonnyPfannschmidt** [comment]
great job, thanks for jumping all the hops to iron that one out

**Pull request merged**